etna/packages/etna-js/actions/magma_actions.js

```diff
@@ -360,7 +360,7 @@ export const requestAnswer = (question, callback = null) => dispatch => {
       else return response;
   }).catch( error => {
     console.error(error);
-    throw new Error(error);
+    return Promise.reject(error);
```

Comment:
Minor nit, but, is there a reason we're converting these throws into Promise.rejects?
As far as I know, throwing an error in promise catch handle is equivalent to returning a Promise.reject, _except for the timing and scheduling_.

Reply:
It does not seem to behave the same, though perhaps it's because I'm using `invoke` incorrectly. If this is not a promise and instead throws an `Error` as the original, then any error thrown causes an exception [here](https://github.com/mountetna/monoetna/blob/master/timur/lib/client/jsx/components/query/query_use_results_actions.tsx#L45). 

```
Error: [object Promise]
    requestAnswer webpack:///./node_modules/etna-js/actions/magma_actions.js?:339
    promise callback*requestAnswer/< webpack:///./node_modules/etna-js/actions/magma_actions.js?:337
    useActionInvoker webpack:///./node_modules/etna-js/hooks/useActionInvoker.js?:22
    runQuery webpack:///./lib/client/jsx/components/query/query_use_results_actions.tsx?:44
    promise callback*useResultsActions/runQuery< webpack:///./lib/client/jsx/components/query/query_use_results_actions.tsx?:42
```

Is there some other way to have the error bubble up to the `catch` block, in that Promise chain?

Reply:
(note that the effect of the error being thrown instead of a promise reject, is a white-screen-of-death in the browser).

Reply:
Hmm... when you say causes an exception, do you mean the web console merely displays the error traceback?
Here's the thing -- your line that you are pointing to, when it calls invoke, it passes through the dispatch which allows that inner call to be invoked, but the catch block there cannot possibly be executing sychronously with the invoke call.  The catch block is called on the promise, providing a callback -- but promise callbacks are never invoked synchronously, they are invoked on next tick according to the spec.

That's what is confusing to me.  The only real difference should just be that Promise.reject also pushes the rejection further away in the scheduling.

regardless of wether a catch block throws a new exception or returns a Promise.reject, it should cascade to the next catch block or cause a console error if no more catch handlers exist.  Something, seems super fishy here, hmm.

Reply:
Actually, there is one difference -- is there a reason you have wraped `error` inside of `new Error(error)`?

You could just use `throw error` here.  In fact, the parameter of a catch callback is equivalent to the value that was originally `throw X`'d.  In this case, your line is simply wrapping the error in more error.

Reply:
Oh, that must be what's going on.

Try changing this line to just `throw error` with no surrounding `new Error(...)` and let me know if that works.

Reply:
Ah, nice, thanks! Yes, that behaves as expected!